File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    stack = []
    
    for i in range(len(s)):
        if len(stack) == 0:
            stack.append(s[i])
        else:
            if stack[-1] == s[i]:
                stack.pop()
            else:
                stack.append(s[i])
    return 0 if stack else 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
    let stack = []
    for (let i=0; i<s.length; i++) {
        if (stack.length === 0) {
            stack.push(s[i])
        } else {
            if (stack[stack.length-1] === s[i]) {
                stack.pop()
            } else {
                stack.push(s[i])
            }
        }
    }
    
    if (stack.length === 0) {
        return 1
    } else {
        return 0
    }
}